## ДЗ по уроку № 6 (05.07.2022)
## Теория вероятности
### Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

**Пример 1**

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np
import pandas as pd
from scipy import stats

In [2]:
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

In [3]:
X = np.array(zp)
Y = np.array(ks)

MX = X.mean()
MY = Y.mean()

cov = ((X - MX) * (Y - MY)).mean()
cov

9157.84

In [4]:
# несмещённая оценка ковариации:

cov1 = ((X - MX) * (Y - MY)).sum() / (X.shape[0] - 1)
cov1

10175.37777777778

In [5]:
cov2 = np.cov(X, Y)
cov2

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [6]:
# коэффициент корреляции Пирсона: 

corr = cov / (X.std() * Y.std())
corr

0.8874900920739162

In [7]:
np.corrcoef(X, Y)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [8]:
df = pd.DataFrame({'zp': zp, 'ks': ks})

df[['zp', 'ks']].corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

**Пример 2**

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [9]:
X = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

p = 0.95

In [10]:
# уровень значимости:

alfa = 1 - p

# размер выборки: 

n = len(X)

# среднее элементов выборки: 

X_ = X.mean()

# среднее квадратичное отклонение в выборке:

sigma = X.std(ddof=1)

In [11]:
# Табличное значение t-критерия Стьюдента для alpha и n

t = stats.t.ppf(1 - alfa / 2, df=n - 1)
t

2.2621571627409915

In [12]:
# Левая граница доверительного интервала: 
    
left = X_ - t * sigma / np.sqrt(n)
left

110.55608365158724

In [13]:
# Правая граница доверительного интервала: 
    
right = X_ + t * sigma / np.sqrt(n)
right

125.64391634841274

Ответ: доверительный интервал (110.5561, 125.6439)

**Пример 3**

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [14]:
D = 25

sigma = D ** 0.5

n = 27

X_ = 174.2

p = 0.95

alfa = 1 - p

In [15]:
# Табличное значение t-критерия Стьюдента для alpha и n

t = stats.t.ppf(1 - alfa / 2, df=n - 1)
t

2.055529438642871

In [16]:
# Левая граница доверительного интервала: 
    
left = X_ - t * sigma / np.sqrt(n)
left

172.2220658754539

In [17]:
# Правая граница доверительного интервала: 
    
right = X_ + t * sigma / np.sqrt(n)
right

176.17793412454608

Ответ: доверительный интервал (172.2221, 176.1779)

**Пример 4**

Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики
и напишите ее в комментарии к Практическому заданию.

Исследование датасета с kaggle по фильмам на IMDb: 

https://www.kaggle.com/datasets/kevalm/movie-imbd-dataset?select=movie_metadata.csv

Планирую рассчитать основные статистические показатели по некоторым признакам: мода, медиана, квартили, дисперсию и прочее пройденное.